src/proxy-ui-api/src/test/java/org/niis/xroad/restapi/openapi/ServicesApiControllerIntegrationTest.java

```diff
@@ -484,22 +484,45 @@ public void addAccessRights() {
         assertEquals(calculatePrimeClientsBefore + 3, updatedServiceClients.size());
     }
 
-    @Test(expected = ConflictException.class)
     @WithMockUser(authorities = { "VIEW_SERVICE_ACL", "EDIT_SERVICE_ACL" })
-    public void addDuplicateAccessRight() {
+    public void addDuplicateAccessRight() throws Exception {
         when(globalConfService.clientsExist(any())).thenReturn(true);
         when(globalConfService.globalGroupsExist(any())).thenReturn(true);
         List<ServiceClient> serviceClients = servicesApiController.getServiceServiceClients(
                 TestUtils.SS1_GET_RANDOM_V1).getBody();
         assertEquals(SS1_GET_RANDOM_SERVICE_CLIENTS, serviceClients.size());
 
+        // add subsystem TestUtils.CLIENT_ID_SS2 as duplicate
         ServiceClients clientsToAdd = new ServiceClients()
                 .addItemsItem(new ServiceClient().id(TestUtils.DB_LOCAL_GROUP_ID_2).serviceClientType(
                         ServiceClientType.LOCALGROUP))
                 .addItemsItem(new ServiceClient().id(TestUtils.CLIENT_ID_SS2).serviceClientType(
                         ServiceClientType.SUBSYSTEM));
+        try {
+            servicesApiController.addServiceServiceClients(TestUtils.SS1_GET_RANDOM_V1, clientsToAdd);
+            throw new Exception("Should throw Conflict exception in stead of this");
```

Comment:
Could use `fail()` assertion here instead of throwing a plain Exception. It would better indicate that there was an actual assertion error and not an application error.

Reply:
Changed.